### ISSUE TITLE
Wrong Active Users Number in Country Ranking

### ISSUE THREAD

**Issue by TomyDoesThings** [ppy/osu-web#9825]
tl;dr
Active users is inaccurately calculated on country leaderboards probably by getting every user that has set at least one ranked score. This is plausible due to how the leaderboards is likely counting the ranked score of every user that has submitted scores, therefore, ACTIVE USERS VALUE IN https://osu.ppy.sh/rankings/(INSERTMODE)/country IS FALSE, which is really unfortunate. Furthermore, there is no way there are 10 million active CTB players. I didn't know osu! would count somebody that's only played once in a game mode as an active player. Many have believed in these values, and I am posting this issue here to say that this should all be fixed.

-----

So, I wanted to see how a percentile system would work in osu! where you grab the ranking you are in a mode and divide by total amount of active players in that mode. As there is no direct number that osu! directly provides you with, I started looking and realized that https://osu.ppy.sh/rankings/osu/country, https://osu.ppy.sh/rankings/taiko/country, https://osu.ppy.sh/rankings/fruits/country, and https://osu.ppy.sh/rankings/mania/country shows the active users in every country.
What does that mean? It means I could add up the active users in each country to get the active users of the whole world, or so I thought...

You see, this is not the first time somebody has thought about what I did. It was done once already around March 10, 2021 using Excel (https://www.reddit.com/r/osugame/comments/m1s442/there_are_15_millions_active_standard_players/), however, I released a program that pulls the live active users from each country live at that moment my program is ran and adds all the active users up in a gamemode by using the active users of each country as a guideline. You can find my project here which also talks about the issue I am going to mention here in this issue: https://github.com/TomyDoesThings/osuUserPercentileFinder/releases/tag/Experimental

As of around 12 PM EST of Feburary 1, 2023, these are the current values of the active users of each mode:
osu! - 22,241,423
Taiko - 11,601,962
CTB - 10,289,321
Mania - 11,282,923

Notice something odd? I did not at first as I did not think this could be a possibility, but **how does a mode like CTB have 10 million active players?** They are not just players. They are **active players**.
To put it into perspective, a player at the time of writing this is at 0.88 PP in CTB. That player goes by the username of Chr0ll0Lucilfer (https://osu.ppy.sh/users/23857539/fruits). **At 0.88 PP in CTB, a player is at global ranking #2,227,932 in that mode.** If there really are 10 million active CTB players, there is approximately 8 million players just like that player who are considered active. That simply is NOT an active player, so something is very wrong here, which leads to me to my next point.

So where is this active users number coming from? Well, me and probably many other osu! players thought it was a live count of all the active players in osu! which by opinion can vary from having set a play within a month or three months, but I just disproved that. I assumed then that this value could perhaps be some false approximation being used, so I thought of a logical way somebody would implement this.
That is when I saw ranked score and **average** ranked score.
By getting ranked score of any country and dividing it by the average score of that same country, you get nearly the same exact active users value for that country. By getting the performance of any country and dividing it by the average performance of that same country, you get nearly the same exact active users value for that country.

As I just proved to you that the active users value must be false, that average score and average performance is certainly also false because they are based off the active users of each country which I have just proven to you are false. Due to how a country's PP is probably added up from all the players that have gotten PP, this makes me think **active users are players that set at least one ranked score in that mode**, which is awfully off from the truth of what an active user/player really is.

We don't have 10M active CTB players (that play monthly) only growing. Even 1.2M active United States CTB players is very suspicious. So many of them are not even playing this game mode, but I will say this, many players do check out what other modes look like and may submit one score.

With this issue I have shown to you, I recommend changing what Active Users really means in https://osu.ppy.sh/rankings/osu/country, https://osu.ppy.sh/rankings/taiko/country, https://osu.ppy.sh/rankings/fruits/country, and https://osu.ppy.sh/rankings/mania/country. They are far from the truth and many have fallen into the trap of actually believing the active users values are true.

I made my program initially to see how good a percentile system would for this game, but with values this false, I can see why the osu!team could maybe a very hesitant on an idea like this, for medals such as the ranking medals.

**Comment by peppy** [comment]
The numbers listed there are from a legacy system which also depends on how often PP recalculations happen.

It's not accurate currently as you have attained.

**Comment by TomyDoesThings** [comment]
I see.

In that case, are there plans to fix this so the information is comfortably usable without such a huge margin of error?

I mean, it's not like this is a rankings issue with the PP rankings. This is information about the playerbase that people have trusted which is not at all accurate (as shown from CTB somehow having 10 million active players). Is there an accurate way to get the real value of all active players and/or players in the game and each individual game mode? If not, shouldn't there be plans to fix this?

To know how many players are active in a specific mode would actually be cool, especially for me testing out the concept of a percentile based ranking system that Ephemeral mentioned of possibly fitting into the ranking medals to make them not exponentially get more difficult for as long as osu! and those medals exist. Those medals are I can see the top (top 50K), The gradual rise (top 10K), Scaling up (top 5K), and Approaching the summit (top 1K).

This feels like an issue significant enough for it to be fixed especially with how most players do not know that these numbers are inaccurate and they're used as data which leads to errors which includes how I tried to get information on if let's say, a #1,000 CTB player in the top 0.1%. Please consider looking into this, @peppy . No pressure though. I know usually you can be pretty busy.

**Comment by peppy** [comment]
This issue will track it getting fixed at some point. You can keep track of it here. No further input is necessary.

**Comment by TomyDoesThings** [comment]
Last thing I will add because I really want this (as can be seen from the Java program I made) is that in order for #3739 (percentile ranking) to happen, it is likely that this issue of incorrect active user count also needs to be fixed.

If those two things can be done, then the ranking medals can also be fixed. I would say that there is nothing but benefits in fixing this issue and getting those things I mentioned implemented. This would benefit the community quite a bit. Not only would those medals finally be fixed, but players can grind for cool percentiles that display their rankings in a more comprehensible way.

You see, a 10K player in osu! has a lot of skill while a 10K player in CTB has barely played the game. The value of PP rankings can only really be seen with how many active players play that game mode. This issue and #3739 are very significant in fixing this overall issue. I hope to see advancements on all of this eventually.

**Comment by TomyDoesThings** [comment]
Hmmm, there was a standard rework recently, yet we still get that there are like 20 million active standard players. I don't believe the data is accurate even after a rework happens. I believe active users counts every not restricted or deleted player that has set at least one score ever. Is there any documentation of this legacy system so we can already think about a fix? @peppy 

I am still very fond of the idea of percentile ranking, and I think this issue being fixed would help a lot with getting to closer to seeing if that's possible (#3739 ).

**Comment by peppy** [comment]
No further comments are necessary here. Please wait patiently until a time comes to look into this, thank you.
